src/java/picard/sam/revertsam/RevertSam.java

```diff
@@ -359,56 +379,19 @@ public boolean filterOut(final SAMRecord first, final SAMRecord second) {
         return out;
     }
 
-    /**
-     * Takes an individual SAMRecord and applies the set of changes/reversions to it that
-     * have been requested by program level options.
-     */
-    public void revertSamRecord(final SAMRecord rec) {
-        if (RESTORE_ORIGINAL_QUALITIES) {
-            final byte[] oq = rec.getOriginalBaseQualities();
-            if (oq != null) {
-                rec.setBaseQualities(oq);
-                rec.setOriginalBaseQualities(null);
-            }
-        }
+    private String getDefaultExtension(final String input) {
```

Comment:
If I want to output by read group, but the input comes from either a URL or `/dev/stdin`, then calling `getDefaultExtension` on `INPUT` will return `.bam`.  Also, this doesn't support `.cram`.  I would add documentation that says the output is BAM by default unless the input ends with `.sam` or .cram`.